templates/nextjs/src/pages/predicate.tsx

```diff
@@ -1,41 +1,36 @@
 import { Button } from "@/components/Button";
 import { FuelLogo } from "@/components/FuelLogo";
 import { Input } from "@/components/Input";
-import { AppContext } from "@/components/Layout";
 import { Link } from "@/components/Link";
+import { useActiveWallet } from "@/hooks/useActiveWallet";
 import { TestPredicateAbi__factory } from "@/sway-api/predicates/index";
 import type { BN, InputValue, Predicate } from "fuels";
 import { BaseAssetId, bn } from "fuels";
-import { useContext, useEffect, useState } from "react";
+import { useState } from "react";
 import toast from "react-hot-toast";
+import useAsync from "react-use/lib/useAsync";
 
 export default function PredicateExample() {
-  const { burnerWallet, burnerWalletBalance, refreshBurnerWalletBalance } =
-    useContext(AppContext);
+  const { wallet, walletBalance, refreshWalletBalance } = useActiveWallet();
 
   const [predicate, setPredicate] = useState<Predicate<InputValue[]>>();
 
   const [predicateBalance, setPredicateBalance] = useState<BN>();
 
   const [pin, setPin] = useState<string>();
 
-  useEffect(() => {
-    (async () => {
-      if (burnerWallet) {
-        const predicate = TestPredicateAbi__factory.createInstance(
-          burnerWallet.provider,
-        );
-        setPredicate(predicate);
-        setPredicateBalance(await predicate.getBalance());
-      }
-
-      // eslint-disable-next-line no-console
-    })().catch(console.error);
-  }, [burnerWallet]);
+  useAsync(async () => {
+    if (wallet) {
```

Comment:
Can `wallet` ever be `undefined`? The `useActiveWallet` hook should guarantee the existence of the wallet.

Reply:
Yeah it will be undefined until it finishes loading. The initialization is async.